src/utils/helpers.ts

```diff
@@ -0,0 +1,50 @@
+interface FutureValueResult {
+    futureValue: number;
+    totalInterest: number;
+    futureValueArray: number[];
+}
+
+export function calculateFutureValue(principal: number, monthlyContribution: number, years: number, yearlyInterestRate: number): FutureValueResult {
+    const futureValueArray: number[] = [];
+
+    // Convert yearly interest rate to decimal form
+    const interestRate = yearlyInterestRate / 100;
+
+    // Calculate the number of compounding periods per year and total number of periods
+    const compoundingFrequency = 12; // Compounded monthly
+    const totalPeriods = years * compoundingFrequency;
+
+    // Calculate future value of initial investment
+    const futureValueInitial = principal * Math.pow(1 + interestRate / compoundingFrequency, totalPeriods);
```

Comment:
the code (and more):
```
Math.pow(1 + interestRate / compoundingFrequency, totalPeriods)
```
is repeated multiple times, so extract to it's own method